crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -294,11 +294,15 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
                     self.handle_room_message_edit(re);
                 }
                 AnyMessageLikeEventContent::RoomMessage(c) => {
-                    self.add(should_add, TimelineItemContent::message(c, relations, self.items));
+                    if should_add {
+                        self.add(TimelineItemContent::message(c, relations, self.items));
+                    }
```

Comment:
Seems much more likely that we introduce bugs by forgetting to check `should_add` if it's not a required parameter of `add`, so I'd prefer not merging this commit.

Reply:
Yeah, I'm not happy about the whole situation: it seems the `if should_add` should happen at most once in the caller's code _above_ the match, but there are callers that will do some unconditional processing before checking `should_add` too. I'll remove this commit and try to find a better way (maybe in a separate PR?).